Description: make sphinx-autogen initialize the sphinx.ext.autodoc module
Author: Jakub Wilk <jwilk@debian.org>
Bug-Debian: http://bugs.debian.org/611078
Bug: https://bitbucket.org/birkenfeld/sphinx/issue/618
Forwarded: yes
Last-Update: 2012-02-04

--- a/sphinx/ext/autosummary/generate.py
+++ b/sphinx/ext/autosummary/generate.py
@@ -27,6 +27,7 @@
 from jinja2 import FileSystemLoader, TemplateNotFound
 from jinja2.sandbox import SandboxedEnvironment
 
+from sphinx.ext import autodoc
 from sphinx import package_dir
 from sphinx.ext.autosummary import import_by_name, get_documenter
 from sphinx.jinja2glue import BuiltinTemplateLoader
@@ -50,6 +51,15 @@
     if len(args) < 1:
         p.error('no input files given')
 
+    class Sphinx(object):
+        def add_autodocumenter(self, cls):
+            autodoc.add_documenter(cls)
+        def add_event(self, name):
+            pass
+        def add_config_value(self, name, default, rebuild):
+            pass
+    autodoc.setup(Sphinx())
+
     generate_autosummary_docs(args, options.output_dir,
                               "." + options.suffix,
                               template_dir=options.templates)
